packages/react/src/hooks/useT.js

```diff
@@ -19,12 +18,12 @@ import {
  * } */
 
 export default function useT(_str, props) {
-  const [translation, setTranslation] = useState('');
+  const setCounter = useState(0)[1];
   useEffect(() => {
-    function render() { setTranslation(t(_str, props)); }
-    render();
-    onEvent(LOCALE_CHANGED, render);
-    return () => { offEvent(LOCALE_CHANGED, render); };
-  }, [_str, props]);
-  return translation;
+    // Using `setCounter` will trigger a rerender
+    function rerender() { setCounter((c) => c + 1); }
+    onEvent(LOCALE_CHANGED, rerender);
+    return () => { offEvent(LOCALE_CHANGED, rerender); };
+  }, [setCounter]);
```

Comment:
Should you wrap this callback function with a `useCallback` to memorize this function?

Also, never seen this syntax `useState(0)[1]`, what are you getting with the `[1]`?

Reply:
There's no "functional" function here, ie a function that deterministically returns a value based on its arguments without having any side-effects. The `onEvent` and `offEvent` are there to register side-effects on the `tx` object. So, I don't see something to memorized.

The `const setCounter = useState(0)[1];` line was originally `const [counter, setCounter] = useState(0);`, but the linter was complaining that `count` wasn't used anywhere. If this was Python, I could have done `_, set_counter = use_state(0)` and be fine (there's a convention in Python that if you are not interested in a return value, you can assign it to `_`). Essentially, I am only using the `counter` state variable as an excuse to rerender the component; I don't care about its value at all, only that it gets changed every time the app's language changes so that React decides to rerender the T-component.

Reply:
@kbairak The setCounter could be set as `[,setCounter] = useState(0);` ;)